sql/core/src/main/java/org/apache/spark/sql/execution/vectorized/WritableColumnVector.java

```diff
@@ -92,17 +92,22 @@ public void reserve(int requiredCapacity) {
       } else {
         throwUnsupportedException(requiredCapacity, null);
       }
+    } else if (requiredCapacity < 0) {
```

Comment:
Although this should work for most cases, is it still possible that the overflowed value can be between `capacity` and `MAX_CAPACITY` ? I know it should be quite rare corner case but just want to point it out whether we care about that.

Reply:
Yea, I also agree that we shall have this simple check merged ASAP.

Reply:
nit: maybe it's cleaner to move this check to the beginning of `reserve`?

Reply:
Ok.